src/CFamily/CMake/CompilationDatabaseLocator.cs

```diff
@@ -86,9 +86,18 @@ public string Locate()
             var cmakeSettingsFullPath = Path.GetFullPath(Path.Combine(rootDirectory, CMakeSettingsFileName));
             var activeConfiguration = buildConfigProvider.GetActiveConfig(rootDirectory);
 
-            return fileSystem.File.Exists(cmakeSettingsFullPath)
+            var compilationDatabaseLocation = fileSystem.File.Exists(cmakeSettingsFullPath)
                 ? GetConfiguredLocation(cmakeSettingsFullPath, activeConfiguration, rootDirectory)
                 : GetDefaultLocation(rootDirectory, activeConfiguration);
+
+            if (fileSystem.File.Exists(compilationDatabaseLocation))
```

Comment:
I didn't want to repeat this logic in telemetry, so it felt more sense to return null if the file does not exist.